crates/task-impls/src/consensus.rs

```diff
@@ -236,6 +235,17 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     self.event_stream
                         .publish(HotShotEvent::QuorumVoteSend(vote))
                         .await;
+                    if let Some((payload_commit, meta)) = &self.payload_commitment_and_metadata {
+                        let (genesis_payload, genesis_meta) =
+                            <TYPES::BlockPayload as BlockPayload>::genesis();
+                        let genesis_commitment = vid_commitment(
+                            &genesis_payload.encode().unwrap().collect(),
+                            self.quorum_membership.total_nodes(),
+                        );
```

Comment:
We are computing an extraneous VID commitment every time we receive a vote? We need to cache this somewhere. Or we could add an explicit `is_genesis` flag to `payload_commitment_and_metadata`, since we should know at the time we set this whether it is the genesis or not

Reply:
This only happens if the proposal is for view 1 and the genesis flag on the justify QC is set.  So it only happens in the first round of voting

Reply:
We could add a flag to remove this check